### ISSUE TITLE
please support regexp

### ISSUE THREAD

**Issue by luckylooke** [Canop/JSON.prune#11]
Thank you for sharing! Could you please support regexps? 

```
JSON.prune({regexp: /abc123/});
"{"regexp":{}}"
```

expecting value same as from `/abc123/.toString()`
```
JSON.prune({regexp: /abc123/});
"{"regexp":"/abc123/"}"
```

Even better if you support also [keyed collections](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects#Keyed_collections), I mostly use maps
```
JSON.prune({map: new Map([
  [1, 'one'],
  [2, 'two'],
  [3, 'three'],
])});
"{"map":{}}"
```
can be stringified by
```
let myMap = new Map([
  [1, 'one'],
  [2, 'two'],
  [3, 'three'],
]);
JSON.stringify(myMap, (key, value) => (value instanceof Map ? [...value] : value));
"[[1,"one"],[2,"two"],[3,"three"]]"
```

Many thanks :)

**Comment by Canop** [comment]
This looks reasonable.

**Comment by GrosSacASac** [comment]
A map can also be cast to an object and back if keys are strings only 

```js
let asObject = Object.fromEntries(myMap);
let map = new Map(Object.entries(asObject));
```
As an alternative to the syntax ```[...value]``` consider ```Array.from``` that accepts an argument to map over keys, values to have a chance to prune them as well.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from

**Comment by luckylooke** [comment]
@GrosSacASac actualy map can have anything as keys.. so maybe it could be implemented somehow safely.. with type check or something.

I was even wondering that functions are cleared from result but there could be referenced somehow... for example you have access to function name if it is defined, so maybe in result there could be some notice that there was function with name "xyz".

I wanted to use this lib to see webpack configuration in my project.. because it is somehow directed by framework... but I have found that vue cli have command for it "vue inspect" so I was able to put it in a file and read (`vue inspect > webpack_cfg.txt`) 💪🙂 

Anyway I find this lib very useful and I am sure I will use it in the future, thank you for sharing! 👍
